plumbing/storer/shallow.go

```diff
@@ -0,0 +1,10 @@
+package storer
+
+import "gopkg.in/src-d/go-git.v4/plumbing"
+
+// ShallowStorer storage of shallow commits, meaning that it does not have the
+// parents of a commit (explanation from git documentation)
+type ShallowStorer interface {
```

Comment:
I'd propose something along these lines:
```
// ShallowStorer storage of references to shallow commits by hash, meaning that these commits have
// missing parents because of a shallow fetch.
```
?